lib/routers/profile/email-preferences.js

```diff
@@ -0,0 +1,45 @@
+const { Router } = require('express');
+const { permissions, whitelist } = require('../../middleware');
+
+const update = () => (req, res, next) => {
+  const params = {
+    id: req.profileId,
+    model: 'emailPreferences',
+    data: {
+      ...req.body.data,
+      profileId: req.profileId
+    }
+  };
+
+  return req.workflow.update(params)
+    .then(response => {
+      res.response = response.json.data;
+      next();
+    })
+    .catch(next);
+};
+
+module.exports = () => {
+  const router = Router();
+
+  router.get('/',
+    permissions('profile.update', req => ({ profileId: req.profileId })),
+    (req, res, next) => {
+      return Promise.resolve()
+        .then(() => req.models.EmailPreferences.query().findOne({ profileId: req.profileId }))
+        .then(emailPreferences => {
+          res.response = emailPreferences;
+        })
+        .then(() => next())
+        .catch(next);
+    }
+  );
+
+  router.put('/',
+    permissions('profile.update', req => ({ profileId: req.profileId })),
```

Comment:
It would be _marginally_ neater to add the permissions check to a single `router.use` so it only needs to be declared once for all methods, but not so much that I won't merge if you don't fancy it.